modules/ding_base/ding_base.module

```diff
@@ -750,3 +757,139 @@ function ding_base_validate_email($email) {
+
+    if (empty($filters['ding_base_term_name'])) {
+      // There is no such filter so we have to add it.
+      $view->add_item(
+        $view->current_display,
+        'filter',
+        'taxonomy_term_data',
+        'name',
+        $filter_options
+      );
+    }
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function ding_base_form_alter(&$form, &$form_state, $form_id) {
+  if ($form_id == 'views_exposed_form' && $form_state['view']->name == 'admin_views_node') {
+    $form['ding_base_term_name']['#type'] = 'textfield';
+    $form['ding_base_term_name']['#id'] = 'edit-ding-base-term-name';
+    $form['ding_base_term_name']['#attached']['js'][] = [
+      'data' => drupal_get_path('module', 'ding_base') . '/js/ding_base.prevent_auto_submit.js',
+      'type' => 'file',
+    ];
+
+    // If no vocabulary selected, disable "Terms" filter.
+    if ($form_state['input']['vid'] == 'All') {
+      $form['ding_base_term_name']['#attributes'] = [
+        'disabled' => TRUE,
+      ];
+    }
+    else {
+      $vid = $form_state['input']['vid'];
+      $form['ding_base_term_name']['#autocomplete_path'] = 'ding_base/taxonomy/autocomplete/' . $vid;
+      $form['ding_base_term_name']['#description'] = t('Autocomplete with terms of selected vocabulary');
+      $form['ding_base_term_name']['#attributes'] = [
+        'class' => [
+          'ctools-auto-submit-exclude',
+        ],
+      ];
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function ding_base_preprocess_views_exposed_form(&$variables) {
+  if (isset($variables['form']) && $variables['form']['#id'] == 'views-exposed-form-admin-views-node-system-1') {
+    $widgets = $variables['widgets'];
+
+    // Get widgets keys for further processing.
+    $widgets_keys = array_keys($widgets);
+
+    // Switch key with value, to have an numeric value.
+    $widgets_keys = array_flip($widgets_keys);
+
+    // Getting Vocabulary filter position.
+    $vid_position = $widgets_keys['filter-vid'];
+    // Set needed position to Taxonomy terms filter field.
+    $widgets_keys['filter-name'] = $vid_position + 1;
+
+    // Modifying positions.
+    foreach ($widgets_keys as $key => $pos) {
+      if ($pos >= $widgets_keys['filter-name'] && $key != 'filter-name') {
+        $widgets_keys[$key] = $pos + 1;
+      }
+    }
+
+    // Flip array to get numeric values as keys.
+    $normalize = array_flip($widgets_keys);
+
+    // Sort array by numeric keys.
+    ksort($normalize, SORT_ASC);
+
+    // Prepare ordered array.
+    $normalized = array_flip($normalize);
+
+    // Order widgets objects by correctly positioned structure.
+    $widgets = array_merge($normalized, $widgets);
+
+    // Return widgets back in variables.
+    $variables['widgets'] = $widgets;
+  }
+}
+
+/**
+ * Admin content "Terms" filter autocomplete callback.
+ */
+function ding_base_taxonomy_autocomplete($vid, $tag) {
+  if (!empty($vid)) {
+    $vocabulary = taxonomy_vocabulary_load($vid);
```

Comment:
Why load the vocabulary here? It's not used in the function!

Reply:
+ My bad. Removed.